modules/cnf-topology-aware-lifecycle-manager-precache-user-spec-images.adoc

```diff
@@ -0,0 +1,181 @@
+    operatorsIndexes:
+      - registry.example.com:5000/custom-redhat-operators:1.0.0
+    operatorsPackagesAndChannels:
+      - local-storage-operator: stable
+      - ptp-operator: stable
+      - sriov-network-operator: stable
+  spaceRequired: 30Gi <2>
+  excludePrecachePatterns:
+    - aws
+    - vsphere
+  additionalImages: <3>
+    - quay.io/exampleconfig/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    - quay.io/exampleconfig/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/exampleconfig/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+----
+<1> The `platformImage`, `operatorsIndexes`, and the `operatorsPackagesAndChannels` fields are automatically populated from the policies of the managed clusters.
+<2> Specifies the minimum required disk space on the cluster in Gigabytes. If unspecified, {cgu-operator} defines a default value for {product-title} images.
+<3> Specifies the list of additional images to pre-cache.
+
+[id="talm-prechache-user-specified-images-preparing-crs_{context}"]
+== Creating the custom resources for pre-caching
+
+You must create the `PreCachingConfig` CR before or concurrently with the `ClusterGroupUpgrade` CR.
+
+. Create the `PreCachingConfig` CR with the list of additional images you want to pre-cache.
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: PreCachingConfig
+metadata:
+  name: exampleconfig
+  namespace: default <1>
+spec:
+  overrides:
+    platformImage: quay.io/openshift-release-dev/ocp-release@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    operatorsIndexes:
+      - registry.example.com:5000/custom-redhat-operators:1.0.0
+    operatorsPackagesAndChannels:
+      - local-storage-operator: stable
+      - ptp-operator: stable
+      - sriov-network-operator: stable
+  spaceRequired: 30Gi
+  excludePrecachePatterns:
+    - aws
+    - vsphere
+  additionalImages:
+    - quay.io/exampleconfig/application1@sha256:3d5800990dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47e2e1ef
+    - quay.io/exampleconfig/application2@sha256:3d5800123dee7cd4727d3fe238a97e2d2976d3808fc925ada29c559a47adfaef
+    - quay.io/exampleconfig/applicationN@sha256:4fe1334adfafadsf987123adfffdaf1243340adfafdedga0991234afdadfsa09
+----
+<1> The `namespace` must be accessible to the hub cluster.
+
+. Create a `ClusterGroupUpgrade` CR with the `preCaching` field set to `true` and specify the `PreCachingConfig` CR created in the previous step:
++
+[source,yaml]
+----
+apiVersion: ran.openshift.io/v1alpha1
+kind: ClusterGroupUpgrade
+metadata:
+  name: cgu
+  namespace: default
+spec:
+  clusters:
+  - sno1
+  - sno2
+  preCaching: true
+  preCachingConfigRef:
+  - name: exampleconfig
+    namespace: default
+  remediationStrategy:
```

Comment:
Missing `managedPolicies` for the platform related images.

Reply:
Got it from [here](https://github.com/openshift-kni/kni-enhancements/blob/main/enhancements/cgu-precaching-additional-images.md#workflow-description) so that might need to be updated as well.

I put this for now, let me know if we should use a different policy name here:
``` 
managedPolicies:
  - talm-policy

Reply:
The [reference](https://github.com/openshift-kni/kni-enhancements/blob/main/enhancements/cgu-precaching-additional-images.md#workflow-description) was an enhancement proposal aimed at show-casing the changes required / proposed. The CGU still needs the managed policies for upgrading the platform-related images (OpenShift and operators). I would suggest using what has been referenced previously in the docs. 

For example:
```yaml
managedPolicies:
  - du-upgrade-platform-upgrade
  - du-upgrade-operator-catsrc-policy
  - common-subscriptions-policy
```